### PR TITLE
ProtectWise - Fixed lastRun to store last event timestamp

### PR DESCRIPTION
**Pull request by omercnet**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19435

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
   - Yes
       - Further details:
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] Dependency 1
- [ ] Dependency 2
- [ ] Dependency 3

## Additional changes
Describe additional changes done, for example adding a function to common server.

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/blob/fix/protectwise-fetch/Integrations/integration-ProtectWise.yml)
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/fix/protectwise-fetch/Integrations/integration-ProtectWise_CHANGELOG.md)

### CODE REVIEW

**Comment by kirbles19** [comment]
@richardbluestone I started the yml, you can take a look at the first few commands/arguments/outputs, and continue. I also did the changelog.md, so just need to finish the yml.

**Comment by omercnet** [comment]
> Looks good, but check with @Itay4 or @yuvalbenshalom for final approval.

thanks Andrew, added @Itay4 for final review

**Comment by richardbluestone** [comment]
Hi I'm going through the yml file have a few questions about 2 commands: 

Observation-search - is defined as "Lookup a single event and its associated observations for ProtectWise". This should be searching for an observation and not an event, as protectwise-event-info searches for events.

protectwise-search-observations - is defined as "search observations in Protectwise".

What is the difference between the 2 commands?

**Comment by omercnet** [comment]
> What is the difference between the 2 commands?

they are the same command, if you look at the code towards the bottom you can see how both cases result in the same code section that handles the request
https://github.com/demisto/content/blob/eb980a4528e40ab5b4ab2292fbc1fd896480a50d/Integrations/integration-ProtectWise.yml#L346-L347

File: Integrations/integration-ProtectWise.yml [hunk 1]
```diff
@@ -203,11 +210,15 @@ script:
                 params.threatCategory, null, null, null, null, null, null, null, token);
             try {
                 var res = JSON.parse(data).events;
+                res.reverse();  # events are fetched in descending order so reversing to process older first
             } catch (ex) {
                 throw 'Error parsing event fetch - ' + data + ' - ' + ex;
             }
             var incidents = [];
             for (var i = 0; i < res.length; i++) {
+                if (incidents.length >= Math.min(parseInt(params.maxFetch), 50)) {
```
Comment (Itay4) [inline Integrations/integration-ProtectWise.yml]: why are we doing `min` here? if the user sets the max fetch to `51`, then it should so

File: Integrations/integration-ProtectWise_CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+## [Unreleased]
+  * Fixed an issue where events were not fetched properly.
```
Comment (Itay4) [inline Integrations/integration-ProtectWise_CHANGELOG.md]: release notes items should start with `-` and not `*`

File: Integrations/integration-ProtectWise.yml (hunk 1 above)
Comment (omercnet) [reply Integrations/integration-ProtectWise.yml]: nope, 50 is a hard limit as done with other integrations until there's a platform solution to handle it properly

Comment (Itay4) [reply Integrations/integration-ProtectWise.yml]: not sure i understand this, but anyway if we want to limit to 50, it should be clear in the parameter display name that max is 50 in anyway

File: Integrations/integration-ProtectWise_CHANGELOG.md (hunk 2 above)
Comment (omercnet) [reply Integrations/integration-ProtectWise_CHANGELOG.md]: fixed

File: Integrations/integration-ProtectWise.yml (hunk 1 above)
Comment (omercnet) [reply Integrations/integration-ProtectWise.yml]: https://github.com/demisto/etc/issues/19251

Comment (Itay4) [reply Integrations/integration-ProtectWise.yml]: im aware of that, but thats not my point
the number `50` is not clear to me, but it doesnt really matter, it should be clear that this is the max so we wont get a bug that someone sets the max to 51 and wont understand why it fetches only 50

Comment (omercnet) [reply Integrations/integration-ProtectWise.yml]: yup, waiting for writers to approve text

**Comment by richardbluestone** [comment]
Hi I've reviewed and updated the yml file. I've added a description where the commands are deprecated. I based it on where the descriptions were identical.

**Comment by Itay4** [comment]
@omercnet the [build](https://circleci.com/gh/demisto/content/30855) fails
image

**Comment by omercnet** [comment]
but

**Comment by Itay4** [comment]
@omercnet build fails:
image

**Comment by Itay4** [comment]
@omercnet build failed:
image

**Comment by Itay4** [comment]
@omercnet can we merge?

**Pull request merged**